nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,344 @@
+> ^\w+{8,10}$
+
+Must contain anything that is not whitespace
+> \S
+
+### Custom Field Choices
+
+The choices to be presented for `Selection` and `Multiple selection` custom field types. These are displayed in the order of the weight values supplied with the lowest weight on top. If regex validation is being used, these choices must match the regular expression.
+
+![Custom Field Choices](./images/custom-fields/custom_field_choices.png)
+
+![Custom Field Select](./images/custom-fields/custom_field_select.png)
+
+## Editing Custom Fields
+
+Since automatic provisioning is only performed when a custom field's content types change, some changes made to existing custom fields are not reflected on the associated objects automatically. Some examples of cases where this might cause unexpected behavior are changes to the `required`, `default` and validation fields. If a custom field is created with `required=False` and then later changed to `required=True`, all of the associated objects will fail validation the next time they're saved unless the custom field is updated with a valid value.
+
+## Deleting Custom Fields
+
+Custom fields are removed from associated objects when a content type is removed from the custom field, including when the custom field is deleted.
+
+## Retrieving Custom Field Data
+
+Custom fields augment an existing model so retrieving custom field values is different from native fields. All custom field data is stored as a dictionary in the model field named `_custom_field_data` but there is a property named `cf` to make accessing this field easier. Example:
+
+### Retrieve Custom Field Data in Nautobot Shell
+
+```py
+# retrieve all custom field data
+>>> device.cf
+{'date_custom_field': '1970-01-01', 'text_custom_field': 'This is the custom field value', 'boolean_custom_field': True, 'integer_custom_field': 12345, 'grouped_custom_field_1': '', 'grouped_custom_field_2': '', 'grouped_custom_field_3': ''}
+
+# retrieve a single field
+>>> device.cf.get("date_custom_field")
+'1970-01-01'
+```
+
+### Retrieve Custom Field Data in the Rest API
+
+Custom fields are returned in the API for all supported models in the `custom_fields` key:
+
+`GET http://localhost:8080/api/dcim/devices/ffd8df99-6d1a-41c3-b19f-b8357eefc481/`
+
+<!-- markdownlint-disable MD033 -->
+<details>
+<summary>View API Results</summary>
+
+```json
+{
+  "id": "ffd8df99-6d1a-41c3-b19f-b8357eefc481",
+  ...
+  "custom_fields": {
+    "boolean_custom_field": true,
+    "date_custom_field": "1970-01-01",
+    "grouped_custom_field_1": "",
+    "grouped_custom_field_2": "",
+    "grouped_custom_field_3": "",
+    "integer_custom_field": 12345,
+    "text_custom_field": "This is the custom field value"
+  }
+}
+```
+
+</details>
+
+### GraphQL
+
+#### Retrieve Data for a Custom Field in GraphQL
+
+Individual custom fields can be retrieved in GraphQL queries by using the `cf_<fieldname>` field name format:
+
+```graphql
+{
+  devices {
+    cf_text_custom_field
```

Comment:
Could we use something a bit more real life over `text_custom_field`? It kinda makes you think that you standard is actually `cf_<fieldname>_custom_field`,